tests/compendia_tests.py

```diff
@@ -90,46 +98,54 @@ class CompendiaTests(unittest.TestCase):
+        result = pyrefinebio.Compendia.search(primary_organism_name="HUMAN")
 
-    def assertCompendia(self, actual, expected):
-        # Assert that feilds returned are properly set and can be accessed by dot notation
-        self.assertEqual(actual.id, expected["id"])
-        self.assertEqual(actual.primary_organism_name, expected["primary_organism_name"])
-        self.assertListEqual(actual.organism_names, expected["organism_names"])
-        self.assertEqual(actual.svd_algorithm, expected["svd_algorithm"])
-        self.assertEqual(actual.quant_sf_only, expected["quant_sf_only"])
-        self.assertEqual(actual.compendium_version, expected["compendium_version"])
-
-        self.assertEqual(actual.computed_file.id, expected["computed_file"]["id"])
-        self.assertEqual(actual.computed_file.filename, expected["computed_file"]["filename"])
-        self.assertEqual(actual.computed_file.size_in_bytes, expected["computed_file"]["size_in_bytes"])
-        self.assertEqual(actual.computed_file.is_smashable, expected["computed_file"]["is_smashable"])
-        self.assertEqual(actual.computed_file.is_qc, expected["computed_file"]["is_qc"])
-        self.assertEqual(actual.computed_file.sha1, expected["computed_file"]["sha1"])
-        self.assertEqual(actual.computed_file.s3_bucket, expected["computed_file"]["s3_bucket"])
-        self.assertEqual(actual.computed_file.s3_key, expected["computed_file"]["s3_key"])
-        self.assertEqual(actual.computed_file.created_at, expected["computed_file"]["created_at"])
-        self.assertEqual(actual.computed_file.last_modified, expected["computed_file"]["last_modified"])
-
-        # Assert that non-returned feilds are set to their default values
-        self.assertIsNone(actual.computed_file.compendia_organism_name)
-        self.assertIsNone(actual.computed_file.compendia_version)
-        self.assertIsNone(actual.computed_file.download_url)
-        self.assertIsNone(actual.computed_file.quant_sf_only)
-        self.assertIsNone(actual.computed_file.result)
-        self.assertIsNone(actual.computed_file.s3_url)
-        self.assertListEqual(actual.computed_file.samples, [])
+        result_list = list(result)
+
+        self.assertObject(result_list[0], compendia_object_1)
+        self.assertObject(result_list[1], compendia_object_2)
+
+        self.assertEqual(len(mock_get.call_args_list), 2)
 
+    @patch("pyrefinebio.http.requests.get", side_effect=mock_get_request)
+    def test_compendia_search_with_invalid_filters(self, mock_get):
+        result = pyrefinebio.Compendia.search(foo="bar")
+
+        result_list = list(result)
+
+        self.assertObject(result_list[0], compendia_object_1)
+        self.assertObject(result_list[1], compendia_object_2)
+
+        self.assertEqual(len(mock_get.call_args_list), 2)
+
+    @patch("pyrefinebio.http.requests.get", side_effect=mock_get_request)
+    def test_compendia_500(self, mock_get):
+        with self.assertRaises(Exception):
```

Comment:
We should return a meaningful exception type with a descriptive/helpful message.